bot/constants.py

```diff
@@ -10,694 +10,728 @@
 their default values from `config-default.yml`.
 """
 import os
-from collections.abc import Mapping
 from enum import Enum
 from pathlib import Path
-from typing import Dict, List, Optional
+from typing import Optional
 
-import yaml
+from pydantic import BaseModel, BaseSettings, Field, root_validator
 
-try:
-    import dotenv
-    dotenv.load_dotenv()
-except ModuleNotFoundError:
-    pass
+# Will add a check for the required keys
 
+env_file_path = Path(__file__).parent.parent / ".env"
+server_env_file_path = Path(__file__).parent.parent / ".env.server"
 
-def _env_var_constructor(loader, node):
-    """
-    Implements a custom YAML tag for loading optional environment
-    variables. If the environment variable is set, returns the
-    value of it. Otherwise, returns `None`.
 
-    Example usage in the YAML configuration:
+class EnvConfig(BaseSettings):
+    class Config:
+        env_file = env_file_path, server_env_file_path,
+        env_file_encoding = 'utf-8'
 
-        # Optional app configuration. Set `MY_APP_KEY` in the environment to use it.
-        application:
-            key: !ENV 'MY_APP_KEY'
-    """
 
-    default = None
+class _Miscellaneous(EnvConfig):
+    debug: str = Field(env="BOT_DEBUG", default="true")
```

Comment:
That's not actually it. This is there to ensure backwards compatibility with the env variables that might be in production.
The values are not the same, as you can see here we don't define a prefix for this specific class, unlike the rest where it's usually `something__`

Since in the old setting this was extracted from a pure env variable called BOT_DEBUG, I thought I'd leave it the way it is.
But if we were to change it, we need to make sure that we define them in the prod config as well. WDYT ?

Reply:
I've accepted the upfront migration as a cost of the upgrade. We should not add confusion/maintenance cost to avoid 10 minutes of manual work. This also applies to nearly every single Field in this file.